src/preconnect.js

```diff
@@ -213,34 +230,63 @@ class PreconnectService {
     if (!this.features_.preload) {
       return;
     }
-    if (opt_preloadAs == 'document' && this.platform_.isSafari()) {
-      // Preloading documents currently does not work in Safari,
-      // because it
-      // - does not support preloading iframes
-      // - and uses a different cache for iframes (when loaded without
-      //   as attribute).
+    // For Safari, do not request a resource if it is not a type that we can
+    // set the "as" attribute for. Safari caches by url+type, so we need to use
+    // the correct type and cannot fall back to "fetch".
+    const isSafari = this.platform_.isIos() || this.platform_.isSafari();
+    if (this.isUnsupportedPreloadAs_(preloadAs) && isSafari) {
```

Comment:
Nit: switch the conditions so that `isUnsupportedPreloadAs_` isn't called needlessly.